### PR TITLE
frontend/fix/#1840: added optional image upload for DL and RC

### PR DESCRIPTION
**Pull request by ashkriti**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description
added optional image upload for DL and RC (for Chennai)

## Motivation and Context
#1840 

## How did you test it?
Tested Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/ComponentConfig.purs [hunk 1]
```diff
@@ -57,7 +57,7 @@ primaryButtonConfig state = let
       , width = MATCH_PARENT
       , height = (V 50)
       , background = Color.black900 
-      , margin = Margin 15 0 15 30
+      , margin = if (state.props.openHowToUploadManual&& (not state.data.cityConfig.uploadRCandDL)) then Margin 15 0 15 10 else Margin 15 0 15 30
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/ComponentConfig.purs:60]: use above `imageUploadCondition` variable 
`imageUploadCondition = state.props.openHowToUploadManual && not state.data.cityConfig.uploadRCandDL`

And remove `--getValueFromConfig "imageUploadOptional" || state.props.isValidState` commented code
It was added for this condition only

File: Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs [hunk 2]
```diff
@@ -147,7 +147,8 @@ view push state =
            , padding( PaddingHorizontal 20 20)
            , margin (MarginBottom 10)
            ]
-           , PrimaryButton.view (push <<< PrimaryButtonAction) (primaryButtonConfig state)]
+           , PrimaryButton.view (push <<< PrimaryButtonAction) (primaryButtonConfig state)
+           , if (state.props.openHowToUploadManual && (not state.data.cityConfig.uploadRCandDL)) then skipButton push state else linearLayout[][]]
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs:151]: Remove unnecessary brackets

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/ComponentConfig.purs [hunk 3]
```diff
@@ -51,7 +51,7 @@ primaryButtonConfig state = let
       }
       , width = MATCH_PARENT
       , background = Color.black900
-      , margin = Margin 15 0 15 30
+      , margin = if (state.props.openHowToUploadManual && (not state.data.cityConfig.uploadRCandDL)) then Margin 15 0 15 10 else Margin 15 0 15 30
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/ComponentConfig.purs:54]: put `state.props.openHowToUploadManual && not state.data.cityConfig.uploadRCandDL` in a variable and remove brackets then use the variable

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — APPROVED** [review]

File: Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs [hunk 4]
```diff
@@ -764,6 +765,19 @@ logoutPopupModal push state =
...
+skipButton :: forall w . (Action -> Effect Unit) -> AddVehicleDetailsScreenState -> PrestoDOM (Effect Unit) w
+skipButton push state =
+  textView $
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  , color Color.black
+  , gravity CENTER
+  , onClick push $ const SkipButton
+  , text $ getString SKIP
+  , textSize FontSize.a_16
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs:777]: Remove This line

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/View.purs [hunk 5]
```diff
@@ -153,7 +153,8 @@ linearLayout
             , padding( PaddingHorizontal 20 20)
             , margin (MarginBottom 10)
             ]
-          , PrimaryButton.view (push <<< PrimaryButtonAction) (primaryButtonConfig state)]
+          , PrimaryButton.view (push <<< PrimaryButtonAction) (primaryButtonConfig state)
+          , if state.props.openHowToUploadManual && not state.data.cityConfig.uploadRCandDL then skipButton push state else linearLayout[][]]
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/View.purs:157]: append the view instead of empty linear layout

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**